From c3c9d9a00023d31501ac49f18928a9a6cc656219 Mon Sep 17 00:00:00 2001
From: Ray Chang <ray.chang@technexion.com>
Date: Tue, 18 Jul 2023 11:35:06 +0800
Subject: [PATCH] FwImage: wifi_mod_para.conf: enable mfg mode for IW416

---
 nxp/wifi_mod_para.conf | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/nxp/wifi_mod_para.conf b/nxp/wifi_mod_para.conf
index 956fa77..8745ad1 100644
--- a/nxp/wifi_mod_para.conf
+++ b/nxp/wifi_mod_para.conf
@@ -137,14 +137,10 @@ SD9098_1 = {
 }
 
 SDIW416 = {
-	cfg80211_wext=0xf
-	drv_mode=0x3
-	max_vir_bss=1
+	mfg_mode=1
+	drv_mode=1
 	cal_data_cfg=none
-	ps_mode=2
-	auto_ds=2
-	host_mlme=1
-	fw_name=nxp/sdiouartiw416_combo_v0.bin
+	fw_name=nxp/sdio8978_uart_combo.bin
 }
 
 SD8801 = {
-- 
2.25.1

