From 0f65263094a22a5c2e621a5f174badcbb3fe2ab7 Mon Sep 17 00:00:00 2001
From: Ray Chang <ray.chang@technexion.com>
Date: Mon, 21 Jul 2025 16:51:41 +0800
Subject: [PATCH] iptables: add sanity check for x6_options in prepare_target()

Upstream-Status: Pending
---
 src/iptables.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/iptables.c b/src/iptables.c
index a81a077..824cbb9 100644
--- a/src/iptables.c
+++ b/src/iptables.c
@@ -2746,6 +2746,11 @@ static struct xtables_target *prepare_target(struct connman_iptables *table,
 
 	switch (table->type) {
 	case AF_INET:
+		// Add a sanity check for x6_options
+		if ((uintptr_t)xt_t->x6_options > 0xFFFFFFFF) {
+			DBG("Warning: x6_options value looks invalid for target %s", xt_t->name);
+			xt_t->x6_options = NULL;
+		}
 		if (xt_t->x6_options)
 			iptables_globals.opts =
 				xtables_options_xfrm(
@@ -2768,6 +2773,11 @@ static struct xtables_target *prepare_target(struct connman_iptables *table,
 
 		break;
 	case AF_INET6:
+		// Add a sanity check for x6_options
+		if ((uintptr_t)xt_t->x6_options > 0xFFFFFFFF) {
+			DBG("Warning: x6_options value looks invalid for target %s", xt_t->name);
+			xt_t->x6_options = NULL;
+		}
 		if (xt_t->x6_options)
 			ip6tables_globals.opts =
 				xtables_options_xfrm(
